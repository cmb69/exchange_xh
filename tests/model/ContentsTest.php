<?php

namespace Exchange\Model;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;

/** @small */
class ContentsTest extends TestCase
{
    private function sut(): Contents
    {
        $sut = new Contents();
        $page1 = $sut->appendPage("Start", ["url" => "\\Start'"], "<h1>Start</h1>");
        $page2 = $page1->appendChild("Sub", [], "<h1>Sub</h1>");
        $page1->appendChild("Sub2", [], "<h1>Sub2</h1>");
        $page2->appendChild("SubSub", [], "<h1>SubSub</h1>");
        $page3 = $sut->appendPage("About", [], "<h1>About</h1>");
        $page3->appendChild("Me", [], "<h1>About me</h1>");
        return $sut;
    }

    public function testSerializesXhString(): void
    {
        Approvals::verifyHtml($this->sut()->toXhString());
    }

    public function testUnserializesXhString(): void
    {
        $expected = $this->sut();
        $actual = Contents::fromXhString($expected->toXhString());
        $this->assertEquals($expected, $actual);
    }
}
