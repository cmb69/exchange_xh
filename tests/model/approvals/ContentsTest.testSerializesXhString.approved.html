<html><head><title>content file</title>
</head><body>
<!--XH_ml1:Start-->
<?php
$page_data[]=array(
'url'=>'\\Start\'',
);
?>
<h1>Start</h1>
<!--XH_ml2:Sub-->
<?php
$page_data[]=array(
);
?>
<h1>Sub</h1>
<!--XH_ml3:SubSub-->
<?php
$page_data[]=array(
);
?>
<h1>SubSub</h1>
<!--XH_ml2:Sub2-->
<?php
$page_data[]=array(
);
?>
<h1>Sub2</h1>
<!--XH_ml1:About-->
<?php
$page_data[]=array(
);
?>
<h1>About</h1>
<!--XH_ml2:Me-->
<?php
$page_data[]=array(
);
?>
<h1>About me</h1>
</body></html>
